src/averageAges.js

```diff
@@ -14,12 +15,27 @@
  *
  * @return {number}
  */
+
+const SEX_MALE = 'm';
+const SEX_FEMALE = 'f';
+const ONE_CENTURY = 100;
+
+function reduceAverage(array) {
+  return array.reduce((personAge, person) => {
+    return personAge + (person.died - person.born) / array.length;
+  }, 0);
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredMen = people.filter(person => {
+    return person.sex !== SEX_MALE
+      ? false
+      : century !== undefined
+        ? Math.ceil(person.died / ONE_CENTURY) === century
+        : true;
+  });
```

Comment:
Don't compare with undefined/null or empty string. Use the logical not to check the falsy values.